engine/gamesys/src/gamesys/scripts/script_resource.cpp

```diff
@@ -842,6 +878,707 @@ static int SetTexture(lua_State* L)
+        case dmGameObject::PLAYBACK_LOOP_FORWARD:  return dmGameSystemDDF::PLAYBACK_LOOP_FORWARD;
+        case dmGameObject::PLAYBACK_LOOP_BACKWARD: return dmGameSystemDDF::PLAYBACK_LOOP_BACKWARD;
+        case dmGameObject::PLAYBACK_LOOP_PINGPONG: return dmGameSystemDDF::PLAYBACK_LOOP_PINGPONG;
+        default:break;
+    }
+    assert(0);
+    return (dmGameSystemDDF::Playback) -1;
+}
+
+static void ValidateSetAtlasAnimations(lua_State* L, uint32_t* num_animations_out)
+{
+    int top = lua_gettop(L);
+    uint32_t num_animations = 0;
+
+    lua_getfield(L, -1, "animations");
+    luaL_checktype(L, -1, LUA_TTABLE);
+
+    lua_pushnil(L);
+    while (lua_next(L, -2))
+    {
+        luaL_checktype(L, -1, LUA_TTABLE);
+        luaL_checkinteger(L, -2);
+
+        // Note: checkstring can change the lua stack, so we use isstring instead
+        lua_getfield(L, -1, "id");
+        if (!lua_isstring(L, -1))
+        {
+            luaL_error(L, "Invalid 'id' in animations table at index [%d], either missing or wrong type", num_animations + 1);
+        }
+        lua_pop(L, 1);
+
+        #define CHECK_LUA_FIELD(check_fn, field_name, required) \
+            lua_getfield(L, -1, field_name); \
+            if (required || !lua_isnil(L,-1)) \
+                check_fn(L, -1); \
+            lua_pop(L, 1);
+
+        CHECK_LUA_FIELD(luaL_checkinteger,      "width", true);
+        CHECK_LUA_FIELD(luaL_checkinteger,      "height", true);
+        CHECK_LUA_FIELD(luaL_checkinteger,      "frame_start", true);
+        CHECK_LUA_FIELD(luaL_checkinteger,      "frame_end", true);
+        CHECK_LUA_FIELD(luaL_checkinteger,      "playback", false);
+        CHECK_LUA_FIELD(luaL_checkinteger,      "fps", false );
+        CHECK_LUA_FIELD(dmScript::CheckBoolean, "flip_vertical", false );
+        CHECK_LUA_FIELD(dmScript::CheckBoolean, "flip_horizontal", false );
+
+        #undef CHECK_LUA_FIELD
+
+        lua_pop(L, 1);
+
+        num_animations++;
+    }
+
+    lua_pop(L, 1);
+    *num_animations_out = num_animations;
+
+    assert(lua_gettop(L) == top);
+}
+
+static void ValidateSetAtlasGeometries(lua_State* L, uint32_t* num_geometries_out)
+{
+    int top = lua_gettop(L);
+    uint32_t num_geometries = 0;
+
+    lua_getfield(L, -1, "geometries");
+    luaL_checktype(L, -1, LUA_TTABLE);
+
+    lua_pushnil(L);
+    while (lua_next(L, -2))
+    {
+        luaL_checktype(L, -1, LUA_TTABLE);
+        luaL_checkinteger(L, -2);
+
+        #define CHECK_NUMBER_ARRAY(field_name) \
+            { \
+                lua_getfield(L, -1, field_name); \
+                luaL_checktype(L, -1, LUA_TTABLE); \
+                lua_pushnil(L); \
+                while (lua_next(L, -2)) \
+                { \
+                    luaL_checkinteger(L, -1); \
+                    luaL_checktype(L, -2, LUA_TNUMBER); \
+                    lua_pop(L, 1); \
+                } \
+                lua_pop(L, 1); \
+            }
```

Comment:
Perhaps get the min/max of the values, to use error checks afterwards?
E.g. making sure that the indices are in the proper range of the vertices/uvs.